sdk/tests/iot/common/test_az_iot_common.c

```diff
@@ -258,6 +258,22 @@ static void test_az_iot_provisioning_client_logging_succeed()
   az_log_set_classifications(NULL);
 }
 
+#ifndef AZ_NO_LOGGING
```

Comment:
Is this `ifndef` needed? I believe the other logging tests work without it by relying on `_az_BUILT_WITH_LOGGING`.